HHS/Head-Start-TTADP
# [TTAHUB-3026] Self Service Automation - Database Backup - backup retention

## Description of change
Implementation of the following as part of the backup process:

- All backups for the 0-30 days (This includes daily backups, deployment triggered backups, manually triggered backups).
- Daily backups for 31-60 days
- Monday daily backups + 1st & 15th of the month backups for for 61-90 days
- 1st of the month for backups older then 90 days

New features to the latest_backup.sh script
- flag based options
- listing of available files
- requesting a specific file
- download and verify
- erase file from server

## How to test

- run "manual-trigger" on the pipeline to trigger backup and backup retention rules be applied
- use latest_backup.sh script to verify all remaining files follow the retention schedule

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3026


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
